tests/test_aiohttp.py

```diff
@@ -122,28 +121,33 @@ async def test_subscribe_disconnect(client: TestClient):
 
 @pytest.mark.asyncio
 async def test_subscribe_pv(ioc: Popen, client: TestClient, subscription_data):
-    results = []
+    results: List[Optional[Dict[str, Any]]] = []
 
     ws_protocol, msg_init, msg_ack, msg_send = subscription_data
 
     async with client.ws_connect("/ws", protocols=[ws_protocol]) as ws:
         await ws.send_json(msg_init)
         response = await ws.receive_json()
         assert response == msg_ack
-        await asyncio.sleep(0.1)
-        start = time.time()
         await ws.send_json(msg_send)
-        await caput(PV_PREFIX + "ticking", 0.0)
+        startVal = 0.0
+        count = 0
         while True:
-            if time.time() - start > 0.5:
+            if count > 2:
                 break
+            # Get the starting value in the subscription for checks later
+            if count == 0:
+                startVal = await caget(PV_PREFIX + "ticking")
```

Comment:
Is there a reason not to move this outside of the `while` loop? Means we can remove the `if` check completely.

Reply:
I tried moving the `caget` outside of the `while` loop. We call this to get the starting value of the PV for the subscription result check. However, moving the `caget` out of the loop caused the tests to fail as the `caget` value did not match the starting value of the subscription. I then moved the `caget` immediately after the value is obtained from the subscription. This performed better but in testing the tests still failed 2/50 times.
Instead I decided to just get the value of the first subscription directly from the subscription result. From there was can check that we obtained the next two and did not miss any. To do this I changed the subscription to include the value as a float which makes parsing easier.